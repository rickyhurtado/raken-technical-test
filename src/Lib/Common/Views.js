import React from 'react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

export const PageTitle = ({ ...props }) => {
  let siteName = process.env.REACT_APP_SITE_NAME;
  let { title, customTitle, appName } = props;

  if (appName) {
    siteName = `${appName} | ${siteName}`;
  }

  let headTitle = customTitle ? title : `${title} | ${siteName}`;

  if (!title) {
    title = siteName;
    headTitle = siteName;
  }

  return (
    <h1 className="page-title">
      <Helmet title={headTitle} />
      {title}
    </h1>
  );
};

export const NavLink = ({ ...props }) => {
  const { count } = { ...props};
  let badge;

  if (count) {
    badge = (
      <span className="notification-badge">{count > 99 ? '99+' : count}</span>
    );
  }

  return (
    <li className={navLinkIsActive({ ...props })}>
      <Link to={props.to} onClick={closeNavbar}>{props.title} {badge}</Link>
    </li>
  );
};

export function viewportDimension() {
  const w = window,
        d = document,
        e = d.documentElement,
        g = d.getElementsByTagName('body')[0],
        width = w.innerWidth || e.clientWidth || g.clientWidth,
        height = w.innerHeight||e.clientHeight||g.clientHeight;

  return { width, height };
}

function navLinkIsActive({ ...props }) {
  return props.path === props.to ? 'active' : '';
}

function closeNavbar() {
  const { width } = viewportDimension();

  if (width < 768) {
    document.getElementById('js-navbar-toggle-btn').click();
  }
}
