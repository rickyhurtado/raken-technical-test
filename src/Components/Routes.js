import React, { Component } from 'react';
import { BrowserRouter, Switch, Redirect } from 'react-router-dom';
import Views from '../Views';
import { ViewRoute } from '../Lib/Common/Routes';

export default class Routes extends Component {
  render() {
    return (
      <BrowserRouter onUpdate={() => window.scrollTo(0, 0)}>
        <Switch>
          <ViewRoute exact path='/' component={() => <Redirect to='/dashboard' />} />
          <ViewRoute exact path='/company' component={Views.Company} />
          <ViewRoute exact path='/dashboard' component={Views.Dashboard} />
          <ViewRoute exact path='/projects' component={Views.Projects} />
          <ViewRoute exact path='/team' component={Views.Team} />
          <ViewRoute exact path='/help' component={Views.Help} />
          <ViewRoute exact path='/account' component={() => <Redirect to='/account/my-profile' />} />
          <ViewRoute exact path='/account/my-profile' component={Views.Account.MyProfile} />
          <ViewRoute exact path='/account/my-projects' component={Views.Account.MyProjects} />
          <ViewRoute exact path='/account/notifications' component={Views.Account.Notifications} />
          <ViewRoute exact path='/account/tasks' component={Views.Account.Tasks} />
          <ViewRoute path='*' component={Views.PageNotFound} />
        </Switch>
      </BrowserRouter>
    );
  }
};
