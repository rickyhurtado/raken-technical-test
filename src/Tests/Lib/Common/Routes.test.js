import React from 'react';
import { BrowserRouter, Redirect } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ViewRoute } from '../../../Lib/Common/Routes';

it('ViewRoute component', () => {
  const component = renderer.create(
    <BrowserRouter>
      <ViewRoute exact path='/' component={() => <Redirect to='/test-page' />} />
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
