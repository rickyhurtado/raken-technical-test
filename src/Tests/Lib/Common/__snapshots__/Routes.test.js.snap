// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRoute component 1`] = `null`;
